book/chapters/chapter4/hyperparameter_optimization.qmd

```diff
@@ -617,6 +617,8 @@ round(c(generalization = as.numeric(generalization),
 
 We find that the insample performance overestimates the model performance, which indicates meta-overfitting to the optimal hyperparameters.
 The out-of-sample performance from nested resampling provides a much better estimate of the `r index('generalization performance')`, despite the nested resampling only accessing 100 data points and not all 1,000,000.
+Note that in many cases, even nested resampling will overestimate the generalization error of the learner.
+This is because the models fit during tuning use only a subset of the data, whereas the final model uses all available observations.`
```

Comment:
```suggestion
```

I don't disagree but in our experiment we underestimate it and I think we say elsewhere that we can never reduce all biases and we always say the intermediate model is an approximation